src/Nova/Booking.php

```diff
@@ -56,63 +52,75 @@ public function fieldsForIndex(NovaRequest $request)
     {
         return array_filter([
             ID::make()->sortable(),
-            Text::make('Room', 'room.id', function () {
-                return $this->room->name;
-            })->sortable(),
-            Text::make('Game Start', 'slot.start_at', function () {
-                return $this->slot->formatted_start;
-            })->sortable(),
-            Date::make('Game Date', 'slot.date')->sortable(),
-            Number::make('Participants'),
-            DateTime::make('Purchased', 'created_at')->sortable(),
+
+            Text::make('Slot Number')->sortable(),
+
+            Number::make('Total Participants')->sortable(),
+
+            MorphTo::make('Experience')->types([
+                nova('escaperoom_game')
+            ]),
+
+            MorphTo::make('Subject')->types([
+                nova('room')
+            ]),
+
+            MorphTo::make('Agent')->types([
+                nova('user')
+            ]),
         ]);
     }
 
     public function fields(Request $request)
     {
         return array_filter([
-            nova('slot') ? BelongsTo::make('Slot', 'slot', nova('slot')) : null,
-            Text::make('Game Start', 'slot.start_at', function () {
-                return $this->slot->formatted_start;
-            })->exceptOnForms(),
-            Number::make('Participants')->exceptOnForms(),
-            Boolean::make('Private Game?', 'is_private')->exceptOnForms(),
-
-            new Panel('Customer', $this->customerFields()),
+            nova('booking_slot') ? BelongsTo::make('Booking Slot', 'bookingSlot', nova('booking_slot')) : null,
+
+            Text::make('Slot Number')
+                ->rules('required')
+                ->sortable(),
+
+            nova('user') ? BelongsTo::make('User', 'user', nova('user'))->sortable() : null,
+
+            nova('location') ? BelongsTo::make('Location', 'location', nova('location'))->sortable() : null,
+
+            Number::make('Total Participants')->rules('required', 'integer', 'min:1'),
```

Comment:
I believe this is supposed to be an array

Reply:
We can pass Rule instance, array, or string, but Psalm doesn't understand that so I am going to use array.

<img width="1076" alt="Screen Shot 2021-05-06 at 9 11 59 PM" src="https://user-images.githubusercontent.com/6707194/117396856-b559c080-aeaf-11eb-8639-1227a7b8b436.png">